File: Data Structures & Algorithms/anagram-groups/submission-3.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        #inputs: array of strings: str
        #outputs: list that contains sublist with every anagram.

        # if using hash map, what are keys? what are values?
        # keys are characters, values are indices?

        results = defaultdict(list)

        for s in strs:
            count = [0] * 26
            for c in s:
                count[ord(c) - ord("a")] += 1
            results[tuple(count)].append(s)
        return results.values()

File: Data Structures & Algorithms/two-integer-sum-ii/submission-3.swift
class Solution {
    func twoSum(_ numbers: [Int], _ target: Int) -> [Int] {
        var (L, R) = (0, numbers.count - 1)
        var setNumbers = Set(numbers)

        while L <= R {
            if numbers[L] + numbers[R] == target {
                return [L + 1, R + 1]
            } else if !setNumbers.contains(target - numbers[L]) {
                L += 1
            } else if !setNumbers.contains(target - numbers[R]) {
                R -= 1
            }
        }
        return []
    }
}
